### PR TITLE
[FEAT] Add Casbin perm enforcer to GSH

### PR DESCRIPTION
**Pull request by Krlier**
This PR aims to add [Casbin][1], a permissions enforcer, into GSH.

[1]:https://casbin.org/

### CODE REVIEW

**Comment by Krlier** [comment]
At the time of writing I'm working on adding a new functionality to the [Casbin code], to identify a policy through an identification string only.

[Casbin code]:https://github.com/casbin/casbin

File: api/handlers/jwt.go [hunk 1]
```diff
@@ -11,56 +11,82 @@ import (
...
+	jwtSlice := strings.Split(authorizationHeader, "JWT")
+	if len(jwtSlice) != 1 {
+		return token, fmt.Errorf("JWT: %s", err.Error())
+	}
+	jwt := jwtSlice[0]
+
+	token, err = parseIDToken(jwt)
 	if err != nil {
-		return fmt.Errorf(err.Error())
+		return token, fmt.Errorf(err.Error())
```
Comment (mdjunior) [inline api/handlers/jwt.go]: If error exists, this return can be:
```
return nil, fmt.Errorf(err.Error())
```

Comment (mdjunior) [reply api/handlers/jwt.go]: I checked this, but this function returns a `IDToken` type. It is not possible returns a `nil`.

**Comment by mdjunior** [comment]
These commits allows add, remove, associate, and disassociate roles in the GSH API.  I added a few pages on the wiki (https://github.com/globocom/gsh/wiki/api-routes#permissions) explaining how new routes work.

**Comment by mdjunior** [comment]
Update page with CLI commands: https://github.com/globocom/gsh/wiki/cli-commands

**Comment by fbomlisboa** [comment]
Great job!

**Pull request merged**